source/Halibut.Tests/Support/BackwardsCompatibility/LatestClientAndPreviousServiceVersionBuilder.cs

```diff
@@ -206,7 +207,10 @@ public async Task<ClientAndService> Build()
                     proxyDetails,
                     halibutLogLevel).Run();
 
-                int listenPort = (int)runningOldHalibutBinary.ServiceListenPort!;
+                var listenPort = (int)runningOldHalibutBinary.ServiceListenPort!;
+
+                portForwarder = portForwarderFactory?.Invoke(listenPort);
```

Comment:
So we weren't actually using the port forwarder when testing a listening connection? That feels like a fairly large thing to have been missed and still have all our tests working?

So what is/was going on here?

Reply:
A lack of tests around the builders.